recipe-generator-node/index.js

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env node
+
+const { ArgumentParser } = require("argparse");
+const { exec } = require("child_process");
+
+const { FILEPATH } = require("./vars");
+
+const REQUIRED_ARGS = [
+    {name: "domain", help: "name of ingredient domain"},
+    {name: "flavour", help: "type of generation to perform; valid options: java-ingredient, java-hook, js-ingredient, js-hook, ts-ingredient, ts-hook"},
+    {name: "cookbook", help: "path to the yaml cookbook definition file"},
+    {name: "targetDir", help: "directory to output generated files"}
+];
+
+const OPTIONAL_ARGS = [
+    {name: "javaPackage", metavar: "package", help: "java package for generated classes"},
+    {name: "ingredientPostfix", metavar: "postfix", help: "string to append to ingredient names"}
+];
+
+// parse required and optional arguments
+const parser = new ArgumentParser({addHelp: true});
+for (let arg of REQUIRED_ARGS) {
+    parser.addArgument(arg.name, {help: arg.help});    
+}
+for (let arg of OPTIONAL_ARGS) {
+    parser.addArgument("--" + arg.name, {help: arg.help, metavar: arg.metavar});    
+}
+const options = parser.parseArgs();
+
+// form the execution string
+let execStr = `java -jar ${FILEPATH} `;
```

Comment:
I guess we're assuming `java` is on the `$PATH` when this script is running? Is that ok? 🤔

Could also try using `$JAVA_HOME`, but it's probably also possible that `java` is on the `$PATH` but `$JAVA_HOME` is undefined. So idk which is best/"correct"...

Reply:
Yeah I'm also unsure. I think I'll leave it as is for now and wait until someone posts an issue.